File: 프로그래머스/BFS&DFS/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    q = [[0, 0]]
    
    while q:
        idx, num = q.pop()
        
        if idx < len(numbers):
            q.append([idx+1, num+numbers[idx]])
            q.append([idx+1, num-numbers[idx]])
        
        if idx == len(numbers):
            if num == target: answer += 1
    
    return answer

File: 프로그래머스/문제_lv2/타겟넘버.js
function solution(numbers, target) {
  var answer = 0;

  const dfs = (cnt, sum) => {
    if (cnt === numbers.length) {
      if (sum === target) {
        answer += 1;
      }
      return;
    }

    dfs(cnt + 1, sum + numbers[cnt]);
    dfs(cnt + 1, sum - numbers[cnt]);
  };

  dfs(0, 0);

  return answer;
}
